install/kubernetes/helm/istio/charts/kiali/values.yaml

```diff
@@ -19,9 +19,13 @@ ingress:
     #   hosts:
     #     - kiali.local
 dashboard:
+  # if not secretname is set kiali fullname will be used
   secretName: kiali
   usernameKey: username
   passphraseKey: passphrase
+  # if no password is set a 40 digit random password will be set
+  username: admin
```

Comment:
We do not want to expose username and password in values.yaml because of security issue, please refer to https://github.com/istio/istio/issues/7447 for detail

Reply:
I think the short answer is you don't do this for either username or password.  We have moved to reference by key names for the passwords and usernames throughout the codebase for security reasons.  If you need passwords specified, doing so in the Helm `values.yaml` file is the wrong place to do it, as it exposes the secrets during helm workflows (via debug and key resolution).

Instead i'd recommend sorting out a workflow which creates your secrets (username and passphrase) via kubectl secret create.

Reply:
We don't store our secrets in the values.yaml directly. We use an additional secrets.yaml (encrypted by SOPS). In additon we're using the Helm secrets plugin. 

To be able to use the chart, without the need to use a SOPS encrypted secrets.yaml (like for CI), we use empty password values which are replaced with random strings by default.

Only our jenkins user is able to decrypt secrets. Devs only have the possbility to encrypt (via Google KMS). So nobody would see a real password while debugging locally. You just get the random string password in helm template / debug. "helm get" would work of cause but i think you could also get it via "kubectl get secret" if you're able to create the secret via kubectl so i'm not sure if there is a real difference security wise?

As our setup is maybe not so common i understand your point and i guess this pr has no chance to be merged, right?

Reply:
Kubernetes secrets can be secured (and encrypted at rest as well).  See https://kubernetes.io/docs/concepts/configuration/secret/#best-practices - list/view for information on locking out kubectl get/list, so there is a delta security wise.

Regarding your setup, Id receomment an "istio-secrets" helm chart which registers your secrets to the defaults you specify for usernameKey and passphraseKey.  This is not secure, but it would work with your workflow.  Once a secret leaks once, it is forever tainted.

With this model the current workflow used with SOPS+helm secret plugin to create a simple chart that only registers the passwords (IOW outside of the Istio  
charts).

We really want Istio to be super secure, and Kubernetes secrets offer the most secure path (hence why the default passwords were removed from the values.yaml - which cannot be encrypted at rest and can be read by multiple system tools, including the console during rendering.